bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -1081,13 +1084,21 @@ public void setLength(long length) {
     }
 
     private void checkFixedLength(long length) {
-        if (arrayType != null && arrayType.getTag() == TypeTags.TUPLE_TAG) {
-            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.INHERENT_TYPE_VIOLATION_ERROR,
-                    RuntimeErrors.CANNOT_CHANGE_TUPLE_SIZE);
+        if (arrayType == null) {
+            return;
```

Comment:
I think we should throw an exception here too. Let's check and do that separately maybe.